modules/core/src/main/java/org/apache/ignite/internal/util/nio/GridTcpNioCommunicationClient.java

```diff
@@ -157,22 +161,24 @@ public GridNioSession session() {
         return S.toString(GridTcpNioCommunicationClient.class, this, super.toString());
     }
 
-    public void sendHeartbeatsIfNeeded() {
-        if (!useHeartbeats)
+    /**
+     * Sends special {@link ConnectionCheckMessage} through the channel to check if connection still alive.
+     */
+    public void checkConnection() {
+        if (!enableConnectionCheckMessage)
             return;
 
-        long sinceLastHeartbeat = U.currentTimeMillis() - ses.lastHeartbeat();
+        long sinceLastConnectionCheck = U.currentTimeMillis() - lastConnectionCheck;
 
-        if (sinceLastHeartbeat > HEARTBEAT_FREQUENCY) {
-            HeartbeatMessage msg = new HeartbeatMessage();
+        if (sinceLastConnectionCheck > CONNECTION_CHECK_MIN_INTERVAL_MS) {
```

Comment:
Do we need this throttling at all? This is triggered by an event that is pretty rare (an incoming connection attempt), so it doesn't seem to make much sense to throttle it, it seems.

Reply:
removed